### ISSUE TITLE
[LC0017] Analyzer crashes with InvalidCastException

### ISSUE THREAD

**Issue by invalidCards** [StefanMaron/BusinessCentral.LinterCop#149]
I am getting the following error after upgrading to the latest version of the LinterCop in one of my projects:
```
[{
	"resource": "/c:/[redacted]/app.json",
	"owner": "_generated_diagnostic_collection_name_#0",
	"code": "AD0001",
	"severity": 4,
	"message": "Analyzer 'BusinessCentral.LinterCop.Design.Rule0017WriteToFlowField' threw an exception of type 'System.InvalidCastException' with message 'System.InvalidCastException: Unable to cast object of type 'Microsoft.Dynamics.Nav.CodeAnalysis.BoundTextIndexAccess' to type 'Microsoft.Dynamics.Nav.CodeAnalysis.IFieldAccess'.
   at BusinessCentral.LinterCop.Design.Rule0017WriteToFlowField.CheckForWriteToFlowField(OperationAnalysisContext context)
   at Microsoft.Dynamics.Nav.CodeAnalysis.Diagnostics.AnalyzerExecutor.<>c__DisplayClass53_1.<ExecuteOperationAction>b__1() in D:\a\1\s\source\Prod\Microsoft.Dynamics.Nav.CodeAnalysis\DiagnosticAnalyzer\AnalyzerExecutor.cs:line 764
   at Microsoft.Dynamics.Nav.CodeAnalysis.Diagnostics.AnalyzerExecutor.ExecuteAndCatchIfThrows_NoLock(DiagnosticAnalyzer analyzer, Action analyze, Nullable`1 info) in D:\a\1\s\source\Prod\Microsoft.Dynamics.Nav.CodeAnalysis\DiagnosticAnalyzer\AnalyzerExecutor.cs:line 1100'",
	"source": "AL",
	"startLineNumber": 1,
	"startColumn": 1,
	"endLineNumber": 1,
	"endColumn": 1
}]
```
Please let me know if you need any files uploaded to troubleshoot this, I can't really make out what the issue is here.

EDIT: It should be noted that this happens with one specific project after upgrading. I cannot reproduce this in any of my other projects.

**Comment by StefanMaron** [comment]
I can reproduce this with that like of code: `Vendor.Name[1] := 'T';`
I am not sure if I can handle that case, I will for now workaround with a try catch

**Comment by StefanMaron** [comment]
Should be fixed in [v0.24.2](https://github.com/StefanMaron/BusinessCentral.LinterCop/releases/tag/v0.24.2)

**Comment by invalidCards** [comment]
I can confirm that the offending project is no longer giving the above warning. Thank you for the quick fix!
